core/templates/dev/head/pages/exploration_player/ImagePreloaderService.js

```diff
@@ -87,14 +87,13 @@ oppia.factory('ImagePreloaderService', [
       AssetsBackendApiService.loadImage(
         ExplorationContextService.getExplorationId(), imageFilename
       ).then(function(loadedImage) {
-        for (var i = 0;
-          i < _filenamesOfImageCurrentlyDownloading.length; i++) {
-          if (_filenamesOfImageCurrentlyDownloading[i] ===
-              loadedImage.filename) {
-            _filenamesOfImageCurrentlyDownloading.splice(i, 1);
-            break;
-          }
-        }
+        _filenamesOfImageCurrentlyDownloading = (
+          _filenamesOfImageCurrentlyDownloading.filter(function(imageFilename) {
+            if (loadedImage.filename !== imageFilename) {
+              return true;
```

Comment:
You can just return `loadedImage.filename !== imageFilename` here

Reply:
Done!